### PR TITLE
Plugins: Add logging to plugin envvars

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

We have the case where we export an environment variable with a '\x00' character leading to a Grafana instance failing to even start.
We don't know which variable it is but suspect it is the client secret.
Adding this simple log line to make sure that if this ever happens we know which env var is faulty for sure.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by dmihai — APPROVED** [review]

**Comment by gamab** [comment]
@grafana/plugins-platform-backend waiting for you approval/reject on this before merging :slightly_smiling_face:

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/services/pluginsintegration/pluginconfig/envvars.go
Comment (andresmgot) [inline pkg/services/pluginsintegration/pluginconfig/envvars.go:182]: I am missing a bit of error handling here. It would help to know what's the error you are seeing. Is it a mistake to set variables with the value `\x00` or is it a problem in the reader of the variable?

If it's not an error, I would set this log level to `Debug` (which still let's you debug the issue if it appears). If it's an error, I would return the error here rather than continuing with the current behavior.

Also, a unit test with what you expect would be nice.

**Review by andresmgot — COMMENTED** [review]
It took me too long to write the comment but in any case, I added my opinion :)

**Pull request merged**
